MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -217,9 +223,43 @@ open class RouteProgress: NSObject {
             }
 
             congestionTravelTimesSegmentsByStep.append(congestionTravelTimesSegmentsByLeg)
+
+            maneuverCoordinateIndex = 0
+            if let segmentMaximumSpeedLimits = leg.segmentMaximumSpeedLimits {
+                for step in leg.steps {
```

Comment:
Consolidate this loop with the loop above that populates the `segmentCongestionLevels` and `expectedSegmentTravelTimes` properties for better performance. We’ll need to refactor a bit of the code, so that the absence of traffic congestion data doesn’t prevent us from processing speed limit data and vice versa.